Fix up types and comments that refer to Deferreds.

clokep opened the pull request:
Some comments refer to functions returning a `Deferred` which have now been converted to async. This tries to clean those up (and some other type information).

This only touches comments and type hints.

richvdh commented on synapse/handlers/identity.py:
```diff
@@ -59,21 +61,23 @@ def __init__(self, hs):
-    async def threepid_from_creds(self, id_server, creds):
+    async def threepid_from_creds(
+        self, id_server: str, creds: Dict[str, str]
+    ) -> Dict[str, Optional[Union[str, int]]]:
```
I think it should be
```suggestion
    ) -> Optional[Dict[str, Union[str, int]]]:
```

Or possibly just
```suggestion
    ) -> Optional[JsonDict]:
```

clokep replied:
Hmm.... yeah the old type was `Deferred[dict[str,str|int]|None]` I think I failed to follow the nested braces properly.

`JsonDict` sounds like a good improvement here. 😄

richvdh commented on synapse/handlers/identity.py:
```diff
@@ -379,9 +386,9 @@ async def requestEmailToken(
-            id_server (str): The identity server to proxy to
-            email (str): The email to send the message to
-            client_secret (str): The unique client_secret sends by the user
+            id_server: The identity server to proxy to
+            email: The email to send the message to
+            client_secret: The unique client_secret sends by the user
```
these should probably go into type annotations rather than just being removed?

richvdh commented on synapse/handlers/identity.py:
```diff
@@ -430,10 +437,10 @@ async def requestMsisdnToken(
-            country (str): The country code of the phone number
-            phone_number (str): The number to send the message to
-            client_secret (str): The unique client_secret sends by the user
+            id_server: The identity server to proxy to
+            country: The country code of the phone number
+            phone_number: The number to send the message to
+            client_secret: The unique client_secret sends by the user
```
ditto

richvdh commented on synapse/handlers/identity.py:
```diff
@@ -485,9 +492,9 @@ async def validate_threepid_session(self, client_secret, sid):
         Tries validating against any configured account_threepid_delegates as well as locally.
 
         Args:
-            client_secret (str): A secret provided by the client
+            client_secret: A secret provided by the client
 
-            sid (str): The ID of the session
+            sid: The ID of the session
```
ditto

richvdh commented on synapse/handlers/identity.py:
```diff
@@ -595,10 +607,10 @@ async def _lookup_3pid_v1(self, id_server, medium, address):
         """Looks up a 3pid in the passed identity server using v1 lookup.
 
         Args:
-            id_server (str): The server name (including port, if required)
+            id_server: The server name (including port, if required)
                 of the identity server to use.
-            medium (str): The type of the third party identifier (e.g. "email").
-            address (str): The third party identifier (e.g. "foo@example.com").
+            medium: The type of the third party identifier (e.g. "email").
+            address: The third party identifier (e.g. "foo@example.com").
```
ok, I'm going to stop saying "ditto" now. What am I missing here?

clokep replied:
Hmm...I'm not sure why I did that, sorry to waste time! I'll take another look.

clokep commented:
Thanks for the review @richvdh! I corrected the missing type hints and re-checked that file to make sure the rest seem sane. I added it to mypy too. 👍
